pages/content/posts/self-experimentation-with-csp.md

```diff
@@ -0,0 +1,51 @@
+
+Next, I found that although both [web.dev](https://web.dev/articles/strict-csp) and [OWASP](https://cheatsheetseries.owasp.org/cheatsheets/Content_Security_Policy_Cheat_Sheet.html#csp-types-granularallowlist-based-or-strict) recommend this style, which they call a _strict CSP_, MDN doesn't even mention it. So we filed an issue and started updating the docs.
+
+## Does OWD need a CSP?
+
+While we were discussing how to add a CSP to the OWD website, [Estelle](https://front-end.social/@estelle) asked a really good question: does our site need a CSP? In our pull request to update the CSP guide, [Hamish](https://github.com/hamishwillee) had a similar comment: [how, specifically, does a CSP protect websites from XSS?](https://github.com/mdn/content/pull/36157#discussion_r1789293865)
+
+Our current documentation tended to go like this:
+
+> 1. XSS is bad
+> 2. CSP protects you from XSS
+> 3. So you should have a CSP
+
+We need to go deeper than this. We ought to explain when and how XSS can cause specific problems for sites, and exactly where a CSP can help: where the features of a CSP meet up with the prongs of an XSS attack. We ought to explain how the threats and corresponding protections might be different for different sites. For example:
+
+- a site where the server dynamically constructs responses from templates, based on the contents of a database
+- a single-page app that uses client-side rendering to update the page content
+- a more or less static site where the server always serves the same content, and the page contains little to no JavaScript.
+
+Our CSP guide update made some steps towards this, but there's more work to do here: in particular we need better documentation of XSS, to cover the different ways an XSS attack can be made and the different tools that can be used to counter it: not only CSP, but tools like [trusted types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) and output encoding.
+
+## Auto-CSP
+
+The Netlify feature also connects with an idea that was mentioned in the [TPAC session on security docs](https://openwebdocs.org/content/posts/swag-at-tpac-anaheim/) and which recently [landed in Angular](https://github.com/angular/angular-cli/pull/28663): tool support to deploy a CSP automatically based on a configuration setting. We'd like to include this in our CSP documentation and again, not just its existence but the specific web app architectures for which it is appropriate.
+
+For example, Netlify's dynamic CSP automatically inserts nonces into all `<script>` tags in the document. This is counter to the guidance for strict CSP, which recommends that you should use a proper templating system to insert nonces, or you will add nonces to script tags that were injected server-side.
+
+So this kind of CSP is appropriate for a web app that's based on client-side rendering, as it protects against client-side script injection.
```

Comment:
FYI: I have trouble understanding the last two points (L45 and 47) of an otherwise very enlightening article

Reply:
Thanks for the comment! I just pushed https://github.com/openwebdocs/owd-website/pull/134/commits/7f790133b5c00c95dfdb58082b5ab51945f9ab64, does this clarify it?

Reply:
much clearer now, indeed - thanks